src/connection/WalletConnectV2.ts

```diff
@@ -33,11 +33,21 @@ export class WalletConnectV2Popup extends WalletConnect {
         showQrModal: qrcode,
         rpcMap: RPC_URLS_WITHOUT_FALLBACKS,
         qrModalOptions: {
+          chainImages: undefined,
+          desktopWallets: undefined,
+          enableExplorer: true,
+          explorerExcludedWalletIds: undefined,
+          explorerRecommendedWalletIds: undefined,
+          mobileWallets: undefined,
+          privacyPolicyUrl: undefined,
+          termsOfServiceUrl: undefined,
           themeMode: darkmode ? 'dark' : 'light',
           themeVariables: {
             '--w3m-font-family': '"Inter custom", sans-serif',
             '--w3m-z-index': Z_INDEX.modal.toString(),
           },
+          tokenImages: undefined,
+          walletImages: undefined,
```

Comment:
The new upstream wc package requires explicit enumeration of all options 🤦

Reply:
Here's the [upstream type](https://github.com/WalletConnect/walletconnect-monorepo/blob/5159f7767e14f503bea4e1718b0cdf3e0931a916/providers/ethereum-provider/src/EthereumProvider.ts#L14)
![image](https://github.com/Uniswap/interface/assets/5773490/7d03fe8b-bdb4-4406-9a03-a272f8cd2955)

I'm wondering now if we could update[ this type ](https://github.com/Uniswap/web3-react/blob/main/packages/walletconnect-v2/src/index.ts#L11-L24)in `@web3-react/walletconnect-v2` to something like the following so we can get the correct optional typing. 
```
/**
 * Options to configure the WalletConnect provider.
 * For the full list of options, see {@link https://docs.walletconnect.com/2.0/javascript/providers/ethereum#initialization WalletConnect documentation}.
 */
export type WalletConnectOptions = Omit<Parameters<typeof WalletConnectProvider.init>[0], 'rpcMap'> & {
  /**
   * Map of chainIds to rpc url(s). If multiple urls are provided, the first one that responds
   * within a given timeout will be used. Note that multiple urls are not supported by WalletConnect by default.
   * That's why we extend its options with our own `rpcMap` (@see getBestUrlMap).
   */
  rpcMap?: { [chainId: number]: string | string[] }
  /** @deprecated Use `rpcMap` instead. */
  rpc?: { [chainId: number]: string | string[] }
}
```